src/components/pages/User/Profile.tsx

```diff
@@ -59,75 +56,81 @@ export default function Profile(): ReactElement {
     };
 
     function changeEditMode(): void {
-        console.log("[PROFILE] changedEditMode");
+        console.log("[Profile] changedEditMode");
         setIsEditing(!isEditing);
     }
 
-    const handleSubmit = async (inputUser: UserInformationInputInterface) => {
-        console.log("[PROFILE] handleSubmit");
-        let newUser: UserInformation = {
-            groups: user.groups,
-            userId: user.userId
+    const handleSubmit = async (userInput: UserInformationInputInterface) => {
+        console.log("[Profile] handleSubmit");
+
+        let updatedUser: UserInformation = {
+            ...user,
+            username: userInput.username
         };
-        if (!inputUser.username) {
+
+        if (userInput.password) {
+            // if the user updated the password
+            const hashedPassword = await hashPassword(userInput.password);
+            updatedUser.password = hashedPassword;
+            updatedUser.confirmationPassword = hashedPassword;
+        } else if (user.username === userInput.username) {
+            // if the new username is the old one show erorr instead of calling the backend
+            // FIXME should we even show something here?
             handleAlertVisibility(
                 DEFAULT_ALERT_DURATION,
                 "danger",
-                "Error: Please choose an username."
+                "Error: No Changes."
             );
             return;
         }
-        newUser["username"] = inputUser.username;
-        if (inputUser.password || inputUser.passwordConfirmation) {
-            if (inputUser.password !== inputUser.passwordConfirmation) {
-                handleAlertVisibility(
-                    DEFAULT_ALERT_DURATION,
-                    "danger",
-                    "Error: Password and password confirmation must match."
-                );
-                return;
-            }
-            if (
-                inputUser.password.match(/\d/) == null ||
-                inputUser.password.match(/[a-z]/) == null ||
-                inputUser.password.match(/[A-Z]/) == null ||
-                notMinStrLength(inputUser.password, MIN_PASSWORD_LENGTH)
-            ) {
-                handleAlertVisibility(
-                    DEFAULT_ALERT_DURATION,
-                    "danger",
-                    "Error: Please pay attention to the notes below the input fields."
-                );
-                return;
-            }
-            newUser["password"] = await hashPassword(inputUser.password);
-            newUser["confirmationPassword"] = newUser["password"];
-        }
 
-        await changeUserInformation(newUser)
+        // trigger api call
+        await changeUserInformation(updatedUser)
             .then((res) => {
                 changeEditMode();
+                // FIXME this does never appear, because we rerender it and this gets lost
```

Comment:
Alerts should be defined global. Would need a lot more refacotring.

Reply:
Y. Dont know why we are not using something like this: https://www.npmjs.com/package/react-notifications